src/Simulation/Simulators/SparseSimulator/Native/flat_hash_map.hpp

```diff
@@ -0,0 +1,1502 @@
+//          Copyright Malte Skarupke 2017.
```

Comment:
Is there an STL version of this? Can it be used?

Reply:
This hash map has better performance than the STL hash map, so it would be better if we can use this one. If we can't, the only functional difference from the STL hash map is the "jump_forward" function [defined here](https://github.com/sam-jaques/qsharp-runtime/blob/748672b5664be9cb3111dc6bbb36d2b9e1620615/src/Simulation/Simulators/SparseSimulator/Native/quantum_hash_map.hpp#L364), which makes the multi-threading faster.